### PR TITLE
fix: use nullpool even for user lookup in the celery

### PR DESCRIPTION
**Pull request by bkyryliuk**
### SUMMARY
Follow up to the https://github.com/apache/incubator-superset/pull/10819

Security manager still relies on the default superset db pool instead of nullpool for celery.

### TEST PLAN
[x] existing unit tests
[x] local & staging env

Error msg:
```
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/thirdparty/superset_python3/superset-cpython-38/lib/superset/tasks/schedules.py", line 204, in create_webdriver
    return WebDriverProxy(driver_type=config["WEBDRIVER_TYPE"]).auth(get_reports_user())
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/thirdparty/superset_python3/superset-cpython-38/lib/superset/tasks/schedules.py", line 208, in get_reports_user
    return security_manager.find_user(config["EMAIL_REPORTS_USER"])
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/thirdparty/superset_python3/pip/flask-appbuilder/flask-appbuilder-cpython-38/lib/flask_appbuilder/security/sqla/manager.py", line 156, in find_user
    self.get_session.query(self.user_model)
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/orm/query.py", line 3432, in one_or_none
    ret = list(self)
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/orm/loading.py", line 100, in instances
    cursor.close()
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.raise_(
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/orm/loading.py", line 80, in instances
    rows = [proc(row) for row in fetch]
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/orm/loading.py", line 80, in <listcomp>
    rows = [proc(row) for row in fetch]
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/orm/loading.py", line 524, in _instance
    tuple([row[column] for column in pk_cols]),
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/orm/loading.py", line 524, in <listcomp>
    tuple([row[column] for column in pk_cols]),
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/engine/result.py", line 681, in _key_fallback
    util.raise_(
  File "/srv/yaps/mounts/superset/superset_build/eb36af9819e03e516ffc7d5b9aeabbf6cb067fe5-1280802881dce54672c405f194757e5f/celery.runfiles/__main__/pip/SQLAlchemy/SQLAlchemy-cpython-38/lib/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
sqlalchemy.exc.NoSuchColumnError: "Could not locate column in row for column 'ab_user.id'"
```

### CODE REVIEW

File: superset/tasks/schedules.py [hunk 1]
```diff
@@ -200,12 +201,21 @@ def _get_url_path(view: str, user_friendly: bool = False, **kwargs: Any) -> str:
...
 
 
-def get_reports_user() -> "User":
-    return security_manager.find_user(config["EMAIL_REPORTS_USER"])
+def get_reports_user(session: Session) -> "User":
+    return (
+        session.query(security_manager.user_model)
+        .filter(
+            func.lower(security_manager.user_model.username)
+            == func.lower(config["EMAIL_REPORTS_USER"])
```
Comment (villebro) [inline superset/tasks/schedules.py:215]: Curious about this, what are the cases where it's needed?

File: superset/tasks/schedules.py [hunk 2]
```diff
@@ -200,12 +201,21 @@ def _get_url_path(view: str, user_friendly: bool = False, **kwargs: Any) -> str:
...
+        )
+        .one_or_none()
```
Comment (villebro) [inline superset/tasks/schedules.py:217]: nit: this is mostly personal preference, but I almost feel we should be able to assume that a reports user exists, and raise if it doesn't, i.e. call `.one()`.

**Review by villebro — APPROVED** [review]
Two minor non-blocking comments

File: superset/tasks/schedules.py (hunk 1 above)
Comment (bkyryliuk) [reply superset/tasks/schedules.py:215]: ah, this is from security manager implementation, wanted them to be aligned, email is case insensitive

File: superset/tasks/schedules.py (hunk 2 above)
Comment (bkyryliuk) [reply superset/tasks/schedules.py:217]: good point, fixed

**Pull request merged**